MOPAC locate_ts test fails with GCC 15.0.1

The Fedora packages of MOPAC fail to build with GCC 15.0.1 included in the development version of Fedora (rawhide). The packages build fine in the release versions.

The build failure can be traced back to a single failing test
```
 18/125 Test  #22: locate_ts ........................***Failed    0.45 sec
          MOPAC Job: "Locate_TS.mop" ended normally on Jan 30, 2025, at 08:42.
Traceback (most recent call last):
  File "/builddir/build/BUILD/mopac-23.0.3-build/mopac-23.0.3/tests/run_test.py", line 33, in <module>
    compare_mopac_out_file(out_line, out_list, ref_line, ref_list, float(argv[3]))
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builddir/build/BUILD/mopac-23.0.3-build/mopac-23.0.3/tests/compare_output.py", line 224, in compare_mopac_out_file
    assert len(ref_list) == len(out_list), f'ERROR: output file size mismatch, {len(ref_list)} vs. {len(out_list)}'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: ERROR: output file size mismatch, 2809 vs. 2653
```

The bug affects at least both MOPAC 23.0.1 and 23.0.3.

--- a/src/reactions/big_swap.F90
+++ b/src/reactions/big_swap.F90
@@ -1473,6 +1473,7 @@ subroutine Locate_TS
     double precision, external :: seconds
     logical :: opend
     character :: num*1
+    converged = .false.
     if (l_ts) then
       line = "TS"
     else if (l_nllsq) then

